src/DiIiS-NA/D3-GameServer/GSSystem/AISystem/Brains/AggressiveNPCBrain.cs

```diff
@@ -107,16 +107,10 @@ public override void Think(int tickCounter)
 
 		protected virtual int PickPowerToUse()
 		{
-			if (PresetPowers.Count > 0)
-			{
-				var randomPower = PresetPowers.PickRandom();
-				
-				// should we try several times or pick random from implemented only powers?
-				if (PowerSystem.PowerLoader.HasImplementationForPowerSNO(randomPower))
-					return randomPower;
-			}
-
-			return -1;
+			var implementedPowers = PresetPowers.Where(PowerSystem.PowerLoader.HasImplementationForPowerSNO);
+			return implementedPowers.TryPickRandom(out var randomPower)
```

Comment:
Here I changed logic a bit. So now we are trying to pick random **implemented** power.
According to the comments in similar places that was initial logic